File: KtvAssistant_android/src/assistant/entity/SongRecord.java
package assistant.entity;

import org.json.JSONObject;

import android.text.TextUtils;
import assistant.task.PCommonUtil;
import assistant.util.ShowLog;

public class SongRecord {
	public String songname = "";
	public String singername = "";
	public int score = 0;
	public String room = "";
	public String time = "";
	
	public SongRecord(JSONObject jsonObject) {

		try {
			if (null != jsonObject) {
				String str = jsonObject.optString("music_name");
				if(!TextUtils.isEmpty(str))
					songname =PCommonUtil.decodeBase64(str);
				
				str = jsonObject.optString("music_singer");
				if(!TextUtils.isEmpty(str))
					singername =PCommonUtil.decodeBase64(str);
				
				score = jsonObject.optInt("top_score");
				
				str = jsonObject.optString("Room_name");
				if(!TextUtils.isEmpty(str))
					room =PCommonUtil.decodeBase64(str);
				
				str = jsonObject.optString("top_date");
				if(!TextUtils.isEmpty(str))
					time =PCommonUtil.decodeBase64(str);
			}

		} catch (Exception e) {
			ShowLog.e("Parser SongInfo failed...please check");
		}

	}
}

File: KtvAssistant_ios/SinaKTVAide/Model/DataStruct/SongRecord.m
//
//  SongRecord.m
//  SinaKTVAide
//
//  Created by 顾利平 on 15/5/13.
//  Copyright (c) 2015年 tiange. All rights reserved.
//

#import "SongRecord.h"
#import "PCommonUtil.h"

@implementation SongRecord

-(instancetype)init
{
    self = [super init];
    
    self.songname = @"";
    self.singername = @"";
    self.score = 0;
    self.room = @"";
    self.time = @"";
    
    return self;
}

+(instancetype) initWithDic:(NSDictionary*)dict
{
    SongRecord* ret = nil;
    
    if(dict)
    {
        ret = [[SongRecord alloc] init];
        if(ret)
        {
            id obj = [dict objectForKey:@"music_name"];
            if(obj)
                ret.songname = [PCommonUtil decodeBase64:obj];
            
            obj = [dict objectForKey:@"music_singer"];
            if(obj)
                ret.singername = [PCommonUtil decodeBase64:obj];
            
            obj = [dict objectForKey:@"top_score"];
            if(obj)
                ret.score = [obj intValue];
            
            obj = [dict objectForKey:@"Room_name"];
            if(obj)
                ret.room = [PCommonUtil decodeBase64:obj];
            
            obj = [dict objectForKey:@"top_date"];
            if(obj)
                ret.time = [PCommonUtil decodeBase64:obj];
        }
    }
    return ret;
}

@end
